components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -80,7 +83,9 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
                         validationResponseDTO.getAuthorizedUser());
 
         String appResideOrgId = getOrganizationIdFromTenantDomain(authorizedUser.getTenantDomain());
-        checkOrganizationIsAllowedToSwitch(appResideOrgId, accessingOrgId);
+        OAuthAppDO oAuthAppDO = (OAuthAppDO) tokReqMsgCtx.getProperty("OAuthAppDO");
```

Comment:
Let's use constants. Is there any constant we can found from oauth / framework repods

Reply:
fixed